Makefile

```diff
@@ -423,6 +423,44 @@ catalog-build-replaces: opm ## Build a catalog image using replace mode
 catalog-push: ## Push a catalog image.
 	$(MAKE) docker-push IMG=$(CATALOG_IMG)
 
+# A valid Git branch from https://github.com/openshift/oadp-operator
+PREVIOUS_CHANNEL ?= oadp-1.2
+
+.PHONY: catalog-test-upgrade
+catalog-test-upgrade: TEMP:= $(shell mktemp -d)
+catalog-test-upgrade: PREVIOUS_OPERATOR_IMAGE?=ttl.sh/oadp-operator-previous-$(GIT_REV):1h
+catalog-test-upgrade: PREVIOUS_BUNDLE_IMAGE?=ttl.sh/oadp-operator-previous-bundle-$(GIT_REV):1h
+catalog-test-upgrade: THIS_OPERATOR_IMAGE?=ttl.sh/oadp-operator-$(GIT_REV):1h
+catalog-test-upgrade: THIS_BUNDLE_IMAGE?=ttl.sh/oadp-operator-bundle-$(GIT_REV):1h
+catalog-test-upgrade: CATALOG_IMAGE?=ttl.sh/oadp-operator-catalog-$(GIT_REV):1h
+catalog-test-upgrade: opm ## Prepare a catalog image with two channels: PREVIOUS_CHANNEL and from current branch
+	cp -r ./ $(TEMP)/current
+	git clone --depth=1 git@github.com:openshift/oadp-operator.git -b $(PREVIOUS_CHANNEL) $(TEMP)/$(PREVIOUS_CHANNEL)
+	cd $(TEMP)/$(PREVIOUS_CHANNEL) && IMG=$(PREVIOUS_OPERATOR_IMAGE) BUNDLE_IMG=$(PREVIOUS_BUNDLE_IMAGE) make bundle && \
+		sed -i '/replaces:/d' ./bundle/manifests/oadp-operator.clusterserviceversion.yaml && \
+		IMG=$(PREVIOUS_OPERATOR_IMAGE) BUNDLE_IMG=$(PREVIOUS_BUNDLE_IMAGE) \
+		make docker-build docker-push bundle-build bundle-push && cd -
+	cd $(TEMP)/current && IMG=$(THIS_OPERATOR_IMAGE) BUNDLE_IMG=$(THIS_BUNDLE_IMAGE) make bundle && \
+		sed -i '/replaces:/d' ./bundle/manifests/oadp-operator.clusterserviceversion.yaml && \
+		IMG=$(THIS_OPERATOR_IMAGE) BUNDLE_IMG=$(THIS_BUNDLE_IMAGE) \
+		make docker-build docker-push bundle-build bundle-push && cd -
+	$(OPM) index add --container-tool docker --bundles $(PREVIOUS_BUNDLE_IMAGE),$(THIS_BUNDLE_IMAGE) --tag $(CATALOG_IMAGE)
+	docker push $(CATALOG_IMAGE)
+	# Run the following command to create the catalog
```

Comment:
I tried to make this part of the command, but make command and EOF were not working together :cry: 

If someone kwons how to automate this part

Reply:
EOF don't work here. You can create a .txt or .yaml for this :)

Alternatively if really needed, this could be hacked into a variable but not worth it.

Reply:
thanks, created a yaml, but had to use an ugly syntax of echo, but at least it is fully automated now